xarray/core/combine.py

```diff
@@ -593,8 +652,9 @@ def combine_by_coords(
 
     Parameters
     ----------
-    datasets : sequence of xarray.Dataset
-        Dataset objects to combine.
+    data_objects : sequence of xarray.Dataset or sequence of xarray.DataArray
```

Comment:
Is renaming a breaking change?

Reply:
It technically is a breaking change - if someone was previously passing `combine_by_coords(datasets=...)` then this change will break their code. But renaming the argument does make sense with this PR. Not sure whether that means this justifies a deprecation cycle?

Reply:
I think it probably does. But only a few extra lines and easy to copy from elsewhere.

Reply:
> I have no idea what justifies a deprecation cycle in your project, or how one is performed.

No worries! (It's very briefly mentioned in our [contributing guide](http://xarray.pydata.org/en/stable/contributing.html#backwards-compatibility), but maybe we should expand that...)

xarray has loads of regular users, and we don't want them to find that downloading a new version of xarray breaks their perfectly good code, even in some minor way. Therefore we normally hold their hand through any changes by warning them in the version before, or possibly by anticipating the way in which we might need to make sure that their old way of using the functions still works temporarily, to give them time to switch to our new way.

In this case the only people who could be affected are people who are currently passing `datasets` as a named argument to `combine_by_coords`, so I think we want to catch that specific possibility and tell them to change the named argument to `data_objects` in future. So perhaps by adding the argument back in with a temporary check like

```python
import warnings

def combine_by_coords(data_objects, ..., datasets=None):

    # TODO remove after version 0.19, see PR4696
    if datasets is not None:
        warnings.warn("The datasets argument has been renamed to `data_objects`. In future passing a value for datasets will raise an error.")
        data_objects = datasets
```

Does that make sense?

Reply:
That makes sense. I'm somewhat concerned about users who ignore the warning (many software projects I've seen generate a lot of warnings), however I don't think there's much that can be done since the method signature will have to change anyway.